src/components/NewCommentForm.tsx

```diff
@@ -1,68 +1,167 @@
+
+    setEmail('');
+    setHasEmailError(false);
+
+    setBody('');
+    setHasBodyError(false);
+  };
+
+  const submit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setIsSending(true);
+
+    const nameError = !name.trim();
+    const emailError = !email.trim();
+    const bodyError = !body.trim();
+
+    setHasNameError(nameError);
+    setHasEmailError(emailError);
+    setHasBodyError(bodyError);
+
+    if (nameError || emailError || bodyError) {
+      setIsSending(false);
+
+      return;
+    }
+
+    const newComment = {
+      name,
+      body,
+      email,
+      postId: post.id,
+    };
+
+    addComment(newComment).finally(() => setIsSending(false));
+    reset();
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form data-cy="NewCommentForm" onSubmit={submit} onReset={reset}>
       <div className="field" data-cy="NameField">
         <label className="label" htmlFor="comment-author-name">
           Author Name
         </label>
 
-        <div className="control has-icons-left has-icons-right">
+        <div
+          className={cn('control has-icons-left', {
+            'has-icons-right': hasNameError,
+          })}
+        >
           <input
+            value={name}
+            onChange={handleNameChange}
             type="text"
             name="name"
             id="comment-author-name"
             placeholder="Name Surname"
-            className="input is-danger"
+            className={cn('input', {
+              'is-danger': hasNameError,
+            })}
           />
 
           <span className="icon is-small is-left">
             <i className="fas fa-user" />
           </span>
 
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
+          {hasNameError && (
+            <span
+              className="icon is-small is-right has-text-danger"
+              data-cy="ErrorIcon"
+            >
+              <i className="fas fa-exclamation-triangle" />
+            </span>
+          )}
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
+        {hasNameError && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Name is required
+          </p>
+        )}
       </div>
 
       <div className="field" data-cy="EmailField">
         <label className="label" htmlFor="comment-author-email">
           Author Email
         </label>
 
-        <div className="control has-icons-left has-icons-right">
+        <div
+          className={cn('control has-icons-left', {
+            'has-icons-right': hasEmailError,
+          })}
+        >
           <input
+            value={email}
+            onChange={handleEmailChange}
             type="text"
```

Comment:
The `type` for the email input should be `email` instead of `text`. This will provide built-in validation for email format and improve user experience.